webapp/src/Service/DOMJudgeService.php

```diff
@@ -1671,4 +1674,42 @@ public function getAllowedLanguagesForContest(?Contest $contest) : array {
             ->getQuery()
             ->getResult();
     }
+
+    public function checkNewVersion(): string|false {
+        if (!$this->config->get('check_new_version')) {
+            return false;
+        }
+        $versionLocalString = explode("/", $this->domjudgeVersion)[0];
+        $patch = "/" . substr($this->domjudgeVersion, 0, strrpos($this->domjudgeVersion, '.')) . ".\d/";
```

Comment:
replace these with `explode` on `.`?

Also verify that there are three elements and silently `return false` if not